enhancements/two-nodes-openshift/2no_template.md

```diff
@@ -0,0 +1,589 @@
+
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+How does this proposal affect MicroShift? For example, if the proposal
+adds configuration options through API resources, should any of those
+behaviors also be exposed to MicroShift admins through the
+configuration file for MicroShift?
+
+### Implementation Details/Notes/Constraints
+
+#### Installation flow
+1. We’ll set up Pacemaker and Corosync on RHCOS using MCO layering.
+   * [TBD extend more]
+2. Install an “SNO like” first node using a second bootstrapped node.
+   * This is somewhat similar what is done in SNO CI in AWS (up until the part the bootstrapped node is removed) and is possible because CEO can distinguish the bootstrapped node as a special use case, thus enabling its removal without breaking the etcd quorum for the remaining node.
+   We should be safe after [MGMT-13586](https://issues.redhat.com/browse/MGMT-13586) which makes the installer wait for the bootstrap etcd member to be removed first before shutting it down.
+3. After the bootstrapped node is removed add it to the cluster as a “regular” node.
+4. Switch CEO to “2NO” mode (where it does not manage etcd) and remove the etcd static pods
+   * [TBD localized/global switch]
+   * This is done because we want to allow simpler maintenance and keeping  some of CEO functionality (defragmentation, cert rotation ect…)
+5. Configure Pacemaker/Corosync to manage etcd/kubelet/cri-o
+6. [TBD storage]
+
+#### Fencing / quorum management
+Fencing quorum is managed by corosync and etcd  will be managed by Pacemaker which will force the etcd quorum.
+
+Here is a node failure example demonstrating that:
+1. Corosync on the survivor (Node1)
+2. Etcd loses internal quorum (E-quorum) and goes read-only
+3. Node1 retains “corosync quorum” (C-quorum) and initiates fencing of Node2
+4. Once fencing is successful Pacemaker will use a fence/resource agent (TBD) which will reschedule the workload from the fenced node
+5. Pacemaker on Node1 forces E-quorum (etcd promotion event)
+6. Cluster continues with no redundancy
+7. … time passes …
+8. Node2 has a persistent failure that prevents communication with Node1
+   * Node2 does not have C-quorum (requires forming a membership with it’s peer)
+   * Node2 does not start etcd or kubelet, remains inert waiting for Node1
+9. Persistent failure on Node2 is repaired
+10. Corosync membership containing both nodes forms
+11. Pacemaker “starts” etcd on Node2 as a follower of Node1
+12. Pacemaker “promotes” etcd on Node2 as full replica of Node1
+13. Pacemaker starts kubelet
+14. Cluster continues with 1+1 redundancy
+
+[Here](#failure-handling) is a  more extensive list of failure scenarios.
+
+#### CEO Enhancement
+1. Requires a new infrastructure type in OpenShift APIs
+2. Replicate the functionality of CEO that is tied to static pods? e.g. Certificate rotation, backup/restore, apiserver etcd endpoint controller
```

Comment:
This say's you're planning to replicate this functionality, but https://github.com/openshift/enhancements/pull/1675/files#diff-0339498404cfe988e6d38c75d19b6e12a7b502fdecdc66c0469fc6408843aea8R248-R250 say's you're building a special CEO mode to retain the ability for it to run this functionality, which is it?

Reply:
```suggestion
2. Make sure that even though CEO will not manage etcd, it will still retain other relevant capabilities (defragmentation, certificate rotation, backup/restore etc...).

Reply:
>which is it?

Good comment, I've tried to better rephrase it.